subiquity/core.py

```diff
@@ -151,6 +151,14 @@ def make_screen(self):
             return s
 
     def run(self):
+        if not os.path.exists(self.opts.autoinstall):
+            from cloudinit import atomic_helper, stages
+            cfg = stages.Init().cfg
+            if 'autoinstall' in cfg:
+                atomic_helper.write_file(
+                    self.opts.autoinstall,
+                    yaml.dump(cfg['autoinstall']),
```

Comment:
Unless your yaml is configured to use SafeDumper, you can use cloud-init's

from cloudinit import safeyaml
safeyaml.dumps(cfg)

Reply:
OK sure, I didn't realize you needed to be safe when _dumping_ yaml!? (and in any case, the yaml contains shell scripts that are going to be executed as root so if someone controls your autoinstall data it's not like they have to be sneaky at all)

Reply:
We've had at least one security review scanning code and scolding for not using safe loaders and dumpers.